tests/integration/tls_tests/test_tls.py

```diff
@@ -81,30 +90,49 @@ async def check_certs_correctly_distributed(ops_test: OpsTest, unit: Unit) -> No
 @pytest.mark.abort_on_fail
 async def test_build_and_deploy(ops_test: OpsTest) -> None:
     """Build and deploy three units of MongoDB and one unit of TLS."""
-    async with ops_test.fast_forward():
-        my_charm = await ops_test.build_charm(".")
-        resources = {"mongodb-image": METADATA["resources"]["mongodb-image"]["upstream-source"]}
-        await ops_test.model.deploy(my_charm, num_units=1, resources=resources, series="jammy")
-        await ops_test.model.wait_for_idle(apps=[DATABASE_APP_NAME], status="active", timeout=2000)
-
-        config = {"generate-self-signed-certificates": "true", "ca-common-name": "Test CA"}
-        await ops_test.model.deploy(
-            TLS_CERTIFICATES_APP_NAME, channel="stable", config=config, series="jammy"
-        )
-        await ops_test.model.wait_for_idle(
-            apps=[TLS_CERTIFICATES_APP_NAME], status="active", timeout=1000
-        )
+    app_name = await get_app_name(ops_test)
+    if app_name:
+        await check_or_scale_app(ops_test, app_name, 1)
+    else:
+        app_name = DATABASE_APP_NAME
+        async with ops_test.fast_forward():
+            my_charm = await ops_test.build_charm(".")
+            resources = {
+                "mongodb-image": METADATA["resources"]["mongodb-image"]["upstream-source"]
+            }
+            await ops_test.model.deploy(my_charm, num_units=1, resources=resources, series="jammy")
+            await ops_test.model.wait_for_idle(apps=[app_name], status="active", timeout=2000)
+
+    tls_app_deployed = False
+    status = await ops_test.model.get_status()
+    for app in ops_test.model.applications:
+        if TLS_CERTIFICATES_APP_NAME in status["applications"][app]["charm"]:
+            logger.debug("Found TLS app name app named '%s'", app)
+            tls_app_deployed = True
+
+    if not tls_app_deployed:
+        async with ops_test.fast_forward():
+            config = {"generate-self-signed-certificates": "true", "ca-common-name": "Test CA"}
+            await ops_test.model.deploy(
+                TLS_CERTIFICATES_APP_NAME, channel="stable", config=config, series="jammy"
```

Comment:
note to self, replace this with self-signed-certificates on https://github.com/canonical/mongodb-k8s-operator/pull/244